core/services/vrf/listener_v2.go

```diff
@@ -621,12 +577,14 @@ func (lsn *listenerV2) processRequestsPerSubBatch(
 				"juelsNeeded", p.juelsNeeded.String(),
 				"maxLink", p.maxLink.String(),
 				"gasLimit", p.gasLimit,
-				"attempts", p.req.attempts)
+				"attempts", p.req.attempts,
+				"remainingBalance", startBalanceNoReserveLink.String(),
+			)
 
 			if p.err != nil {
 				if startBalanceNoReserveLink.Cmp(p.juelsNeeded) < 0 && errors.Is(p.err, errPossiblyInsufficientFunds{}) {
-					ll.Infow("Insufficient link balance to fulfill a request based on estimate, returning", "err", p.err)
-					return processed
+					ll.Infow("Insufficient link balance to fulfill a request based on estimate, breaking", "err", p.err)
+					break
```

Comment:
If it breaks here won't it still try to process the next batch, even if out of funds?

Reply:
Ah yep, as the code reads it'll try but sims will continuously fail until we run out of requests to process in the sub. Maybe a better way would be set a flag here and to break out of the outer loop after the current batch has been enqueued?

Reply:
That will reduce the log spam in that case I think.

Reply:
Right yeah, I think we need to break out here to avoid the old behavior where we get DDOSd by underfunded requests, aside from just the log spam

Reply:
Absolutely correct. Done.